tools/SampleAnalyzer/Commons/DataFormat/WeightCollection.h

```diff
@@ -39,137 +39,205 @@
+
+        /// Size
+        MAuint32 size() { return weights_.size(); }
+
+        /// Add a new weight group
+        MAbool Add(MAuint32 id, MAfloat64 value)
         {
-          std::stringstream str;
-          str << id;
-          std::string idname;
-          str >> idname;
-          throw EXCEPTION_WARNING("The Weight '" + idname +
-                                      "' is defined at two times. Redundant values are skipped.",
-                                  "", 0);
+            // Try to add the item
+            std::pair<std::map<MAuint32, MAfloat64>::iterator, bool> ret;
+            ret = weights_.insert(std::pair<MAuint32, MAfloat64>(id, value));
+
+            // Is it added?
+            try
+            {
+                if (!ret.second)
+                {
+                    std::stringstream str;
+                    str << id;
+                    std::string idname;
+                    str >> idname;
+                    throw EXCEPTION_WARNING("The Weight '" + idname +
+                                                "' is defined at two times. Redundant values are skipped.",
+                                            "", 0);
+                }
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                return false;
+            }
+
+            return true;
         }
-      }
-      catch (const std::exception &e)
-      {
-        MANAGE_EXCEPTION(e);
-        return false;
-      }
-
-      return true;
-    }
-
-    /// Get all the Weight Collection
-    const std::map<MAuint32, MAfloat64> &GetWeights() const
-    {
-      return weights_;
-    }
 
-    /// Get a weight
-    const MAfloat64 &Get(MAuint32 id) const
-    {
-      // Try to get the item
-      std::map<MAuint32, MAfloat64>::const_iterator it = weights_.find(id);
-      try
-      {
-        if (it == weights_.end())
+        /// Get all the Weight Collection
+        const std::map<MAuint32, MAfloat64> &GetWeights() const { return weights_; }
+
+        /// Get a weight
+        const MAfloat64 &Get(MAuint32 id) const
         {
-          std::stringstream str;
-          str << id;
-          std::string idname;
-          str >> idname;
-          throw EXCEPTION_ERROR("The Weight '" + idname +
-                                    "' is not defined. Return null value.",
-                                "", 0);
+            // Try to get the item
+            std::map<MAuint32, MAfloat64>::const_iterator it = weights_.find(id);
+            try
+            {
+                if (it == weights_.end())
+                {
+                    std::stringstream str;
+                    str << id;
+                    std::string idname;
+                    str >> idname;
+                    throw EXCEPTION_ERROR("The Weight '" + idname +
+                                              "' is not defined. Return null value.",
```

Comment:
Replace `Return null value` by `A null value is returned`.

Reply:
```suggestion
                                              "' is not defined. A null value is returned.",
```